openupgradelib/openupgrade.py

```diff
@@ -2367,3 +2368,80 @@ def set_xml_ids_noupdate_value(env, module, xml_ids, value):
         SET noupdate = %s
         WHERE module = %s AND name in %s
     """, (value, module, tuple(xml_ids),))
+
+
+def convert_to_company_dependent(
+    env,
+    model_name,
+    origin_field_name,
+    destination_field_name,
+    model_table_name=None,
+):
+    """ For each row in a given table, the value of a given field is
+    set in another 'company dependant' field of the same table.
+    Useful in cases when from one version to another one, some field in a
+    model becomes a 'company dependent' field.
+
+    This method must be executed in post-migration scripts after
+    the field is created, or in pre-migration if you have previously
+    executed add_fields openupgradelib method.
+
+    :param model_name: Name of the model.
+    :param origin_field_name: Name of the field from which the values
+                              will be obtained.
+    :param company_dependent_field_name: Name of the 'company dependent'
+                                         field where the values obtained
+                                         from origin_field_name will be
+                                         set.
+    :param model_table_name: Name of the table.
+    """
+    logger.debug("Converting {} in {} to company_dependent field {}.".format(
+        origin_field_name, model_name, destination_field_name))
+    cr = env.cr
+    mapping_type2field = {
```

Comment:
Check if the source and destination field are different. If not, raise an error saying that a field can't be converted to property without changing its name.